src/core/lib/security/transport/security_handshaker.c

```diff
@@ -227,24 +227,26 @@ static grpc_error *on_handshake_next_done_locked(
     return grpc_set_tsi_error_result(
         GRPC_ERROR_CREATE_FROM_STATIC_STRING("Handshake failed"), result);
   }
-  // Send data to peer.
+  // Update handshaker result.
+  if (handshaker_result != NULL) {
```

Comment:
Why check this value twice, once here and then again on line 243 below?

I think we can just structure this code like this:

```
if (bytes_to_send_size > 0) {
  // write to peer
} else if (handshaker_result != NULL) {
  // call check_peer()
} else {
  // read from peer
}
```